src/mixins/allowable-actions-mixin.ts

```diff
@@ -32,8 +33,14 @@ export default class AllowableActionsMixin extends Vue {
       }
 
       case AllowableActions.DISSOLVE_COMPANY: {
-        return (!this.isHistorical && !this.hasBlocker && !!businessId &&
+        const isDissolveAllowed = (!this.isHistorical && !!businessId &&
           !!getFeatureFlag('supported-dissolution-entities')?.includes(this.getEntityType))
+        // if its not SP/GP , then consider hasBlocker flag (existing)
+        if (!this.isFirm) {
+          return isDissolveAllowed && !this.hasBlocker
+        }
+
+        return isDissolveAllowed
       }
```

Comment:
Looks good.

Since you're now familiar with this code, I should tell you that this was intended to be an "initial implementation". This is business logic and should be in the BE instead. The idea is that the API would return, say, an array of what actions are allowed. However, this is not been fully implemented yet. If you have any thoughts on this, speak up or create a ticket or whatever. I'd still like this business logic to exist in the BE only.

PS Please update the unit tests for this mixin to verify your changes.

Reply:
Absolutely I agree with you, business logic should be in BE